docs/src/testing/test-collection.md

```diff
@@ -1,95 +1,111 @@
+See [Scarb documentation](https://docs.swmansion.com/scarb/docs/reference/targets.html#auto-detection-of-test-targets)
+for more details about the mechanism.
 
-If there is a `lib.cairo` file in `tests` folder,
-then it is treated as an entrypoint to the `tests` package from which tests are collected.
+## Tests Organization
 
-For example, for a package structured this way:
+Test can be placed in both `src` and `test` directories. When adding tests to files in `src` you must wrap them in tests
+module.
 
-```shell
-$ tree .
-.
-├── Scarb.toml
-├── tests/
-│   ├── lib.cairo
-│   ├── common/
-│   │   └── utils.cairo
-│   ├── common.cairo
-│   ├── test_contract.cairo
-│   └── not_included.cairo
-└── src/
-    └── lib.cairo
-```
+You can read more about tests organization
+in [Scarb documentation](https://docs.swmansion.com/scarb/docs/extensions/testing.html#tests-organization).
+
+### Unit Tests
 
-with `tests/lib.cairo` content:
+Test placed in `src` directory are often called unit tests.
+For these test to function in snforge, they must be wrapped in a module marked with `test` attribute.
 
 ```rust
-mod common;
-mod test_contract;
+// src/example.rs
+// ...
+
+// This test is not in module marked with `#[cfg(test)]` so it won't work
+#[test]
+fn my_invalid_test() {
+    // ...
+}
+
+#[cfg(test)]
+mod tests {
+    // This test is in module marked with `#[cfg(test)]` so it will work
+    #[test]
+    fn my_test() {
+        // ..
+    }
+}
 ```
 
-and `tests/common.cairo` content:
+### Integration Tests
+
+Integration tests are placed in `tests` directory.
+This directory is a special directory in Scarb.
+Tests do not have to be wrapped in `#[cfg(test)]` and each file is treated as a separate module.
 
 ```rust
-mod utils;
+// tests/example.rs
+// ...
+
+// This test is in `tests` directory
+// so it works without being in module with `#[cfg(test)]` 
+#[test]
+fn my_test_1() {
+    // ..
+}
 ```
 
-tests from `tests/lib.cairo`, `tests/test_contract.cairo`, `tests/common.cairo`
-and `tests/common/utils.cairo` will be collected.
-
-### Without `tests/lib.cairo`
-
-When there is no `lib.cairo` present in `tests` folder, 
-all test files **directly** in `tests` directory (i.e., not in its subdirectories)
-are treated as modules and added to a single virtual `lib.cairo`. 
-Then this virtual `lib.cairo` is treated as an entrypoint to the `tests` package from which tests are collected.
+#### Modules and `lib.cairo`
 
-For example, for a package structured this way:
+As written above, each file in `tests` directory is treated as a separate module
 
 ```shell
-$ tree .
-.
-├── Scarb.toml
-├── tests/
-│   ├── common/
-│   │   └── utils.cairo
-│   ├── common.cairo
-│   ├── test_contract.cairo
-│   └── not_included/
-│       └── ignored.cairo
-└── src/
-    └── lib.cairo
+$ tree
+tests/
+├── module1.cairo <-- is collected
+├── module2.cairo <-- is collected
+└── module3.cairo <-- is collected
```

Comment:
Please split this shell snippet, similarly as in [this PR](https://github.com/foundry-rs/starknet-foundry/pull/2658) (just to maintain consistency 😄 ).

Btw, I think here we should make the output expanded by default.